src/ds/ds5/ds5-device.cpp

```diff
@@ -692,6 +695,27 @@ namespace librealsense
                 std::make_shared<thermal_compensation>(_thermal_monitor,thermal_compensation_toggle));
             }
 
+            auto const CUR_GVD_VERSION = gvd_buff[gvd_version_offset];
+            auto const MIN_GVD_VERSION = 0x4;
+
+            auto ir_filter_mask = d400_caps::CAP_IR_FILTER;
+            if (MIN_GVD_VERSION <= CUR_GVD_VERSION &&
+                val_in_range(_pid, {RS435_RGB_PID, RS435I_PID, RS455_PID}) &&
+                (_device_capabilities & ir_filter_mask) == ir_filter_mask)
+            {
+                device_name += "F";
+            }
+
+            auto ip65_mask = d400_caps::CAP_IP65;
+            if (MIN_GVD_VERSION <= CUR_GVD_VERSION &&
+                val_in_range(_pid, { RS455_PID }) &&
+                (_device_capabilities & ip65_mask) == ip65_mask)
+            {
+                device_name = device_name.substr(0, device_name.size() - 1);
+                device_name += "6";
```

Comment:
The "F" is OK since it's the last character,
Here we replace so I wish for something more elegant..
Look at std::string function or std::regex variations
And like I said we are missing comments, hard to understand what is done here

Reply:
I added some comments and left commented lines of code with the purpose to understand what looks good for us.